src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1688,7 +1688,7 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
                 if module_candidate is None or not isinstance(module_candidate, str):
                     continue
 
-                candidate_file = os.path.join(component, module_candidate + ".py")
+                candidate_file = f"{component}/{module_candidate}.py"
```

Comment:
`os.path.join()` should automatically adapt to the directory separator conventions of the operating system it's running on. So, not fully sure here.

Reply:
The problem is that paths in `filenames` are not using backslashes.

Reply:
even on windows. I'm using win10 22H2

Reply:
What are they using? That's strange.

Reply:
forward slash actually. I'm not sure if it's rfilename's side effect

Reply:
https://github.com/huggingface/diffusers/blob/6f58f2f979b572479fcea57714156a1b9aec122c/src/diffusers/pipelines/pipeline_utils.py#L1677 this one

Reply:
Pinging @Wauplin maybe.

Reply:
Yes @PENGUINLIONG is right here. `os.path.join` is useful when dealing with local filesystem. Here we want to compute a filepath on the Hub which uses the unix convention (e.g. `path/to/file.txt`). Good catch!
(but I wonder why we did not realize before though... :thinking:).

---

(nit) I would add a small comment for the future ourselves here.

```suggestion
                # We compute candidate file path on the Hub. Do not use `os.path.join`. 
                candidate_file = f"{component}/{module_candidate}.py"
```

Reply:
Thanks, Lucain! The custom module component here is fairly new, probably that's why.

Reply:
Could we maybe use https://docs.python.org/3/library/os.path.html#os.path.normpath